xc/domain.ml

```diff
@@ -55,6 +55,14 @@ type arch_domainconfig = Xenctrl.arch_domainconfig =
 type domain_create_flag = Xenctrl.domain_create_flag =
   | CDF_HVM
   | CDF_HAP
+  | CDF_S3_INTEGRITY
+  | CDF_OOS_OFF
+  | CDF_XS_DOMAIN
+  | CDF_IOMMU
+[@@deriving rpcty]
+
+type domain_create_iommu_opts = Xenctrl.domain_create_iommu_opts =
```

Comment:
Since options are represented by a list, the type of an individual option could be singular: `domain_create_iommu_opt`. But I don't think this matters.

Reply:
It depends on the changes in Xen, if that's what you mean. I don't know what 'xcp-idl' is.

Reply:
There's a possibility there may be more options in future.